### PR TITLE
Leaderboard: hit the API

### PR DESCRIPTION
**Pull request by usmansbk**
In this PR, I implemented the following:

- Creating a new game using the API
- Refresh button to fetch latest scores
- Submit new score entry

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
Hello @usmansbk 

You've done pretty well so far!
But for a few issues, you are almost done with this. Hang in there!


## Required Changes ♻️
_Please check the comments under the review._
- [ ] Ensure you use arrow functions rather than the `function` keyword.

Happy coding, Cheers! 🚀🚀

Do feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether it is valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/util.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    },
+  });
+
+  return response.json();
+}
+
+export async function get(URL) {
+  const response = await fetch(`${BASE_URL}${URL}`);
+  return response.json();
+}
```
Comment (mikenath223) [inline src/util.js]: - [x] Kindly change all these to `es6 arrow functions`. It keeps the project in line with recent trends.

File: src/api.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export async function fetchScores(id) {
+  const { result } = await get(`games/${id}/scores/`);
+  return result;
+}
+
+export async function submitScore(id, data) {
+  const { result } = await post(`games/${id}/scores/`, data);
+  return result;
+}
```
Comment (mikenath223) [inline src/api.js]: - [x] Also please, change these to `es6 arrow functions`.

File: src/util.js (hunk 1 above)
Comment (usmansbk) [reply src/util.js]: @mikenath223 I don't understand why this is blocking my PR from being approved. It's not part of the general requirement and Microverse best practice.

File: src/api.js (hunk 2 above)
Comment (usmansbk) [reply src/api.js]: @mikenath223 I don't understand how this is blocking my PR from being approved. It's not part of the general requirement, and also not part of the Microverse best practice.

**Comment by mikenath223** [comment]
Hi @usmansbk, 

You made a great effort in this project, but unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that will be invalidated for this reason:
-  Kindly ensure you have new commits after the previous review.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Happy coding. Cheers!🚀🚀

If something isn't 100% clear, you can leave comments or questions in this PR thread.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, whether valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **don't count** into the code reviews limit._

**Comment by usmansbk** [comment]
@mikenath223 I'm not changing my code, as I believe it meets the requirements. There's a big difference between Optional and Mandatory requirements. Please give professional feedback as to why I need to change my functions to arrow functions.

**Comment by mikenath223** [comment]
Kindly check this:
ice_screenshot_20210726-094152

**Comment by usmansbk** [comment]
@mikenath223 the project requirements here is a bit different from yours https://github.com/microverseinc/curriculum-javascript/blob/main/leaderboard/m2_send_receive_data.md

**Review by mimipeshy — CHANGES_REQUESTED** [review]
## Hi @usmansbk 👋🏾 👋🏾 , 

Good job on the work you have done so far in this milestone 👏 
 ## Required Changes ♻️

- [ ] The previous TSE requested that you change the use of arrow functions rather than the `function` keyword. If you check just above your `project requirements,` you will find `Javascript requirements`. If you open the link and go to point 4, it encourages the use of ES6. Javascript's ES6 brings new syntax and awesome features to make your code more modern and readable. It allows you to write less code and do more. ES6 introduces us to many great features like arrow functions. The arrow function is really awesome and makes your code more readable, more structured, and look like modern code. Take a minute and also go through this [article on why ES6 arrow functions are so awesome](https://zendev.com/2018/10/01/javascript-arrow-functions-how-why-when.html). Then work on the feedback given 😄 
 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by usmansbk** [comment]
@mimipeshy I've already fixed it. Why did you require the same changes again?

**Review by mikenath223 — APPROVED** [review]
Hello there  

Your project is complete!  There is nothing else to say other than... it's time to merge it 🛳 
Congratulations! 🎉 

Happy coding. Cheers! 🚀🚀

If you feel something isn't clear enough you can leave a question or comment in the PR thread.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SpaYco — COMMENTED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

The project got approved! so nothing left for me to do except inform CRs of situations like this to improve the quality of code reviews :)

Keep rocking 🚀!

**Pull request merged**